modules/nw-metallb-configure-vrf-bgppeer.adoc

```diff
@@ -0,0 +1,183 @@
+
+. Create a `BGPPeer` custom resources (CR).
+
+.. Create a file, such as `frrviavrf.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: frrviavrf
+  namespace: metallb-system
+spec:
+  myASN: 100
+  peerASN: 200
+  peerAddress: 192.168.130.1
+  vrf: ens4vrf <1>
+----
+<1> Defines the VRF instance to associate with the BGP peer. MetalLB can advertise services and make routing decisions based on the routing information in the VRF. 
+
+.. Apply the configuration for the BGP peer by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f frrviavrf.yaml
+----
+
+. Create an `IPAddressPool` CR.
+
+.. Create a file, such as `first-pool.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: first-pool
+  namespace: metallb-system
+spec:
+  addresses:
+  - 192.169.10.0/32
+----
+
+.. Apply the configuration for the IP address pool by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-pool.yaml
+----
+
+. Create a `BGPAdvertisement` CR.
+
+.. Create a file, such as `first-adv.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: first-adv
+  namespace: metallb-system
+spec:
+  ipAddressPools:
+    - first-pool
+  peers:
+    - frrviavrf <1>
+----
+<1> In this example, MetalLB advertises a range of IP addresses from the `first-pool` IP address pool to the `frrviavrf` BGP peer. 
+
+.. Apply the configuration for the BGP advertisement by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f first-adv.yaml
+----
+
+. Create a `Namespace`, `Deployment` and `Service` CR.
+
+.. Create a file, such as `deploy-service.yaml`, with content like the following example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: test
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: server
+  namespace: test
+spec:
+  selector:
+    matchLabels:
+      app: server
+  template:
+    metadata:
+      labels:
+        app: server
+    spec:
+      containers:
+      - name: server
+        image: registry.redhat.io/rhel9/httpd-24 
+        ports:
+        - name: http
+          containerPort: 30100
+        command: ["/bin/sh", "-c"]
+        args: ["sleep INF"] 
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: server1
+  namespace: test
+spec:
+  ports:
+  - name: http
+    port: 30100
+    protocol: TCP
+    targetPort: 30100
+  selector:
+    app: server
+  type: LoadBalancer
+----
+
+.. Apply the configuration for the namespace, deployment, and service by running the follow command:
++
+[source,terminal]
+----
+$ oc apply -f deploy-service.yaml
+----
+
+.Verification
+
+. Get the external IP address for the service by running the following command:
++
+[source,bash]
+----
+$ oc get service -n test <1>
```

Comment:
The right way to verify this part is to enter in the frr container of the speaker pods and check the output for
```
vtysh -c "show bgp vrf xxx neigh
```
and see if the connection is in established state.

Also, running `vtysh -c "show bgp vrf xxx ipv4` to check if the service is advertised correctly.

Reply:
I need some help with example output for the second command as I am not able to replicate this workflow in full.